stan/math/prim/mat/fun/autocovariance.hpp

```diff
@@ -39,6 +39,35 @@ void autocovariance(const std::vector<T>& y, std::vector<T>& acov,
   }
 }
 
+/**
+ * Write autocovariance estimates for every lag for the specified
+ * input sequence into the specified result using the specified
+ * FFT engine.  The return vector be resized to the same length as
+ * the input sequence with lags given by array index.
+ *
+ * <p>The implementation involves a fast Fourier transform,
+ * followed by a normalization, followed by an inverse transform.
+ *
+ * <p>An FFT engine can be created for reuse for type double with:
+ *
+ * <pre>
+ *     Eigen::FFT<double> fft;
+ * </pre>
+ *
+ * @tparam T Scalar type.
+ * @param y Input sequence.
+ * @param acov Autocovariance.
+ * @param fft FFT engine instance.
+ */
+template <typename T, typename DerivedA, typename DerivedB>
+void autocovariance(const Eigen::MatrixBase<DerivedA>& y,
+                    Eigen::MatrixBase<DerivedB>& acov, Eigen::FFT<T>& fft) {
+  autocorrelation(y, acov, fft);
+
+  T var = (y.array() - y.mean()).square().sum() / y.size();
+  acov = acov.array() * var;
```

Comment:
This should be all one one line, with just 
```
acov = acov.array() * ((y.array() - y.mean()).square().sum() / y.size());
```
for optimal use of expression templates and avoid the copy into `var`.